### DISCUSSION TITLE
Use client-go to access multi-cluster pods , but throws some errors

### DISCUSSION THREAD

**Discussion by wuyingjun-lucky** [clusterpedia-io/clusterpedia#213] [Q&A]
<img width="1845" alt="image" src="https://user-images.githubusercontent.com/16109961/172755154-27f85bd6-2351-4c62-9cdb-b4c6bd6bc6a6.png">

**Comment by wuyingjun-lucky** [comment]
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/16109961/172755244-9dd225f7-ce80-434d-8c0c-e97b3f401b9e.png">
<img width="825" alt="image" src="https://user-images.githubusercontent.com/16109961/172755280-922c3c13-3190-49ef-b6f9-83292db8ac86.png">

**Comment by wuyingjun-lucky** [comment]
`var err error
	var config *rest.Config
	var kubeconfig *string


	if home := homedir.HomeDir(); home != "" {
		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "相对路径")
	} else {
		kubeconfig = flag.String("kubeconfig", "", "相对路径")
	}

	if config, err = rest.InClusterConfig(); err!= nil {
		if config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig); err != nil {
			err.Error()
		}
	}

	clientSet, err := kubernetes.NewForConfig(config)
	if err != nil {
		panic(err.Error())
	}

	factory := informers.NewSharedInformerFactory(clientSet, 0)
	deployPods := factory.Core().V1().Pods()
	deployLister := deployPods.Lister()

	stopChan := make(chan struct{})
	defer close(stopChan)

	factory.Start(stopChan)
	factory.WaitForCacheSync(stopChan)


	pods, err := deployLister.Pods("test0608").List(labels.Everything())
	if err != nil {
		panic(err.Error())
	}
	for idx, pod := range pods {
		fmt.Printf("%d ---- %s\n ", idx +1, pod.Name)
		fmt.Printf("%d ---- %v\n ", idx +1, pod.Status.ContainerStatuses)
		fmt.Printf("%d ---- %v\n ", idx +1, pod.Status.String())
	}

	<- stopChan`

**Comment by wuyingjun-lucky** [comment]
@Iceber When I change to cluster30 kubeconfig It works fine .
Can you help me ?

**Comment by Iceber** [comment]
@wuyingjun-lucky Do you want to use clusterpedia to watch multiple cluster resources?

clusterpedia currently only supports retrieval of resources, i.e. get and list operations, not watch operations

**Reply by wuyingjun-lucky** [reply]
We have many applications ，It seems there will be watch operations exist. I will check these days.

**Reply by Iceber** [reply]
If you need to watch a single cluster, I think it is better to connect to the cluster directly. Of course, in the future clusterpedia will support more operations for a single cluster, such as watch, exec, logs, but this will also bring additional proxy loss.

If you need to watch multiple cluster resources, this is a very complex feature, because it involves the apiserver to be aware of resource changes in the case of multiple replicas, we can discuss the implementation further.

**Reply by wuyingjun-lucky** [reply]
Yeah 
Our final goal is to ignore member clusters.
